packages/@aws-cdk/cdk/lib/cloudformation/parameter.ts

```diff
@@ -90,6 +100,8 @@ export class Parameter extends Referenceable {
     super(parent, name);
     this.properties = props;
     this.value = new Ref(this);
+    this.valueAsString = this.value.toString();
+    this.valueAsList = this.value.toList();
```

Comment:
Am always somewhat annoyed that only one of those is valid for a given entity... And we allows both. But I guess we can't do better, can we?

Reply:
I was thinking that we can also provide `StringParameter` and `StringListParamater` as sugar classes that will make this a bit more strongly-typed.